### PR TITLE
Feature/regarch 83 timeline component

### PR DESCRIPTION
**Pull request by igor-kamil**
https://jira.sng.sk/browse/REGARCH-83

Screenshot 2020-07-01 at 15 05 26

### CODE REVIEW

File: resources/js/components/timeline.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+             $("#year_until").val(values[handle]);
+         } else {
+             $("#year_from").val(values[handle]);
+         }
+     });
+
+    timeline.on('end', function( values, handle ) {
+         // var form = $('#filters form');
+         // form.submit();
+         $("#year_until").change();
```
Comment (igor-kamil) [inline resources/js/components/timeline.js:40]: this should be done in nicer way

File: app/Models/Building.php [hunk 2]
```diff
@@ -237,16 +237,32 @@ public static function getFilterValues($payload)
...
+            } elseif (isSet($results['buckets'])) {
+                $values[$attribute] = collect($results['buckets'])
+                ->mapWithKeys(function ($bucket) {
+                    return [$bucket['key'] => $bucket['doc_count']];
+                });
+            } else {
+                $values[$attribute] = collect([]);
+            }
         }
         return $values;
```
Comment (eronisko) [inline app/Models/Building.php:267]: I'm kinda having a hard time parsing this. Would this work?

```php
        return [
            'architects' => collect($searchResult['aggregations']['architects']['buckets'])
                ->flatMap(fn ($bucket) => [$bucket['key'] => $bucket['doc_count']]),

            'functions' => collect($searchResult['aggregations']['functions']['buckets'])
                ->flatMap(fn ($bucket) => [$bucket['key'] => $bucket['doc_count']]),

            'year_min' => $searchResult['aggregations']['year_min']['value'],
            'year_max' => $searchResult['aggregations']['year_max']['value'] + 10,
        ];
```

File: resources/js/components/timeline.js (hunk 1 above)
Comment (eronisko) [reply resources/js/components/timeline.js:40]: Are you planning on changing this? I don't get what's wrong with it.

File: app/Http/Controllers/BuildingController.php [hunk 3]
```diff
@@ -20,9 +20,27 @@ public function index(Request $request)
         }
 
         $filter_values = Building::getFilterValues($buildings->buildPayload());
+
+        if (is_numeric($filter_values['year_max'])) {
+            $filter_values['year_max'] += 10;
+        }
+
```
Comment (eronisko) [inline app/Http/Controllers/BuildingController.php:27]: I'd rather thake care of this in Building::getFilterValues

File: app/Models/Building.php (hunk 2 above)
Comment (igor-kamil) [reply app/Models/Building.php]: this is beautiful! much nicer

File: app/Models/Building.php [hunk 4]
```diff
@@ -237,16 +237,32 @@ public static function getFilterValues($payload)
...
                 ]
             ],
+            'year_min' => [
+                'min' => [
+                    'field' => 'year_from',
+                ]
+            ],
+            'year_max' => [
+                'max' => [
+                    'field' => 'year_from',
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply app/Models/Building.php:247]: it would. do we want it?

**Pull request merged**